libvast/vast/system/source.hpp

```diff
@@ -69,14 +69,17 @@ template <class Reader>
 struct source_state {
   // -- member types -----------------------------------------------------------
 
+  using source_actor = caf::stateful_actor<source_state>;
+
   using factory_type = table_slice_builder_ptr (*)(record_type);
 
   using downstream_manager
     = caf::broadcast_downstream_manager<table_slice_ptr>;
 
   // -- constructors, destructors, and assignment operators --------------------
 
-  source_state(caf::scheduled_actor* selfptr) : self(selfptr) {
+  source_state(caf::scheduled_actor* selfptr)
+    : self{caf::actor_cast<source_actor*>(selfptr)} {
```

Comment:
Is this actually compiling? `actor_cast` is meant for converting between handle types and not for downcasting pointers. Did you try simply taking `source_actor*` as constructor argument? If that doesn't work then please use `static_cast` for downcasting. Or simply revert this set of changes (see previous comment).